.github/workflows/build-commit.yaml

```diff
@@ -0,0 +1,65 @@
+name: Build a Daft commit and store the outputted wheel in AWS S3
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    secrets:
+      ACTIONS_AWS_ROLE_ARN:
+        description: The ARN of the AWS role to assume
+        required: true
+    inputs:
+      commit:
+        type: string
+        description: The commit hash to build
+        required: true
+    outputs:
+      wheel:
+        description: The wheel file that was built
+        value: ${{ jobs.build-commit.outputs.wheel }}
+
+jobs:
+  build-commit:
+    runs-on: buildjet-8vcpu-ubuntu-2004
+    timeout-minutes: 15 # Remove for ssh debugging
+    permissions:
+      id-token: write
+      contents: read
+    outputs:
+      wheel: ${{ steps.build_and_upload.outputs.wheel }}
+    steps:
+    - uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-west-2
+        role-to-assume: ${{ secrets.ACTIONS_AWS_ROLE_ARN }}
+        role-session-name: daft-performance-comparisons-build
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.commit || github.sha }}
+        fetch-depth: 1
+    - uses: ./.github/actions/install
+    - uses: buildjet/cache@v4
+      with:
+        path: ~/target
+        key: ${{ runner.os }}-cargo-deps-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: ${{ runner.os }}-cargo-deps-
+    - name: Build the python wheel and upload it to AWS S3
+      id: build_and_upload
+      run: |
+        if ! ls ~/target/wheels/*.whl 1> /dev/null 2>&1; then
+          # Build wheel
+          export CARGO_TARGET_DIR=~/target
+          uv v
+          source .venv/bin/activate
+          uv pip install pip maturin
+          maturin build --release
+        fi
+
+        # Upload wheel
+        # (there should only be one output wheel in this directory)
```

Comment:
If this is a strong assertion, maybe we can instead `ls` it and grab the first wheel? Otherwise we might run into weird cases where we override the upload with a bunch of wheels in a for loop.

Reply:
Updated.